src/main/kotlin/Delete.kt

```diff
@@ -0,0 +1,38 @@
+import com.mongodb.client.model.Filters
+import com.mongodb.kotlin.client.coroutine.MongoDatabase
+import kotlinx.coroutines.runBlocking
+import java.util.regex.Pattern
+
+fun main() {
+    val db = setupConnection()
+
+    runBlocking {
+        deleteRestaurant(db)
+        deleteRestaurants(db)
+    }
+
+}
+
+suspend fun deleteRestaurant(db: MongoDatabase) {
+    val collection = db.getCollection<Restaurant>(collectionName = "restaurants")
+
+    val queryParams = Filters.regex(Restaurant::name.name, Pattern.compile("^Insert"))
```

Comment:
From @johnlpage : Why are you using $regex in a simple tutorial - we don't really advise using $regex in most cases.

Reply:
@johnlpage: The object created earlier follows a particular naming pattern/

Reply:
So? Still not good practise - what are you trying to show, why not use a range?